component/gnap/as/client_test.go

```diff
@@ -0,0 +1,549 @@
+		},
+		{
+			name:     "error requesting gnap access with empty request",
+			signer:   &mockSigner{SignatureVal: []byte("signature")},
+			tokenVal: "test Success Value",
+			grantReq: nil,
+			errMsg:   "requestAccess: empty request",
+		},
+		{
+			name:     "error requesting gnap access with invalid signer",
+			signer:   &mockSigner{SignatureErr: fmt.Errorf("signing error")},
+			tokenVal: "test Success Value",
+			grantReq: &gnap.AuthRequest{
+				AccessToken: []*gnap.TokenRequest{},
+				Client:      &gnap.RequestClient{},
+				Interact:    &gnap.RequestInteract{},
+			},
+			errMsg: "requestAccess: signature error: signing error",
+		},
+		{
+			name:   "error requesting gnap access with http server returning 501 error",
+			signer: &mockSigner{SignatureVal: []byte("signature")},
+			grantReq: &gnap.AuthRequest{
+				AccessToken: []*gnap.TokenRequest{},
+				Client:      &gnap.RequestClient{},
+				Interact:    &gnap.RequestInteract{},
+			},
+			errMsg: "requestAccess: Auth server replied with invalid Status [/gnap/auth]: 501 Not Implemented",
+		},
+		{
+			name:   "error requesting gnap access with bad http client error",
+			signer: &mockSigner{SignatureVal: []byte("signature")},
+			grantReq: &gnap.AuthRequest{
+				AccessToken: []*gnap.TokenRequest{},
+				Client:      &gnap.RequestClient{},
+				Interact:    &gnap.RequestInteract{},
+			},
+			errMsg: "requestAccess: failed to post HTTP request to [/gnap/auth]:",
+		},
+		{
+			name:   "error requesting gnap access with bad response unmarshall",
+			signer: &mockSigner{SignatureVal: []byte("signature")},
+			grantReq: &gnap.AuthRequest{
+				AccessToken: []*gnap.TokenRequest{},
+				Client:      &gnap.RequestClient{},
+				Interact:    &gnap.RequestInteract{},
+			},
+			errMsg: "requestAccess: read response not properly formatted [/gnap/auth, unexpected end of JSON input]",
+			grantResp: &gnap.AuthResponse{
+				InstanceID: "mocking empty response",
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		tc := tt
+
+		t.Run(tc.name, func(t *testing.T) {
+			hf := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				err := processPOSTAuthAccessRequest(w, r, tc.grantResp)
+				require.NoError(t, err)
+			})
+
+			server, url, httpClient := CreateMockHTTPServerAndClient(t, hf)
+
+			switch tc.name {
+			case "error requesting gnap access with http server returning 501 error":
+				server, url, httpClient = CreateMockHTTPServerAndClientNotOKStatusCode(t, http.StatusNotImplemented)
+			case "error requesting gnap access with bad http client error":
+				httpClient = &http.Client{}
+			case "error requesting gnap access with invalid server URL":
+				url = "bad url"
+			}
+
+			defer func() {
+				e := server.Close()
+				require.NoError(t, e)
+			}()
+
+			c, err := NewClient(tc.signer, httpClient, url)
+			require.NoError(t, err)
+
+			response, err := c.RequestAccess(tc.grantReq)
+			if tc.errMsg != "" {
+				if tc.errMsg[len(tc.errMsg)-1] == ':' {
```

Comment:
Instead of doing this, can't we just update expected message and have one generic validation here ?
ie, `require.EqualError(t, err, tc.errMsg)`

Reply:
updated.. this case has the server url which has a dynamic port, it can't be hardcoded in the error message.

I've updated the case to inject the url in the error message here instead